[BUG] `legiond` compilation fails under kernel 6.14.6: type mismatch involving `struct sigev`

**Problem Description**
Compilation fails with commit `52ced0d` under Linux kernel 6.14.6 (the current version on Arch) due to a type mismatch between the defined `timer_handler()` function and the expected handler format for a `struct sigevent`.

Relevant compiler output:
```
cc -march=native -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -flto=auto -Wall -linih -c legiond.c -o legiond.o
cc -march=native -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -flto=auto -Wall -linih -c legiond-ctl.c -o legiond-ctl.o
legiond.c: In function ‘main’:
legiond.c:77:35: error: assignment to ‘void (*)(__sigval_t)’ {aka ‘void (*)(union sigval)’} from incompatible pointer type ‘void (*)(void)’ [-Wincompatible-pointer-types]
   77 |         sev.sigev_notify_function = timer_handler;
      |                                   ^
legiond.c:32:6: note: ‘timer_handler’ declared here
   32 | void timer_handler()
      |      ^~~~~~~~~~~~~
cc -march=native -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -flto=auto -Wall -linih legiond-ctl.o -o legiond-ctl
make: *** [Makefile:19: legiond.o] Error 1
```

I assume this is caused by a recent-ish change to the kernel, since `legiond.c` hasn't been touched in several months.

Please let me know if I can help troubleshoot/test further!

Edit: `<details>` elements on GitHub apparently don't play nice with codeblocks, so here: `struct sigevent` definition, via `man sigevent` on my system, for ease of reference.
```c
#include <signal.h>

struct sigevent {
    int             sigev_notify;  /* Notification type */
    int             sigev_signo;   /* Signal number */
    union sigval    sigev_value;   /* Data passed with notification */

    typeof(void (union sigval))  *sigev_notify_function;
                                   /* Notification function
                                      (SIGEV_THREAD) */
    pthread_attr_t *sigev_notify_attributes;
                                   /* Notification attributes */

    /* Linux only: */
    pid_t           sigev_notify_thread_id;
                                   /* ID of thread to signal
                                      (SIGEV_THREAD_ID) */
};

union sigval {                     /* Data passed with notification */
    int             sival_int;     /* Integer value */
    void           *sival_ptr;     /* Pointer value */
};
```
</details>

--- a/extra/service/legiond/legiond.c
+++ b/extra/service/legiond/legiond.c
@@ -29,7 +29,7 @@ void term_handler(int signum)
 	exit(0);
 }
 
-void timer_handler()
+void timer_handler(union sigval sigev_value)
 {
 	pretty("config reload start");
 	parseconf(&config);

